dask/dataframe/io/parquet/core.py

```diff
@@ -80,12 +83,28 @@ def __getitem__(self, key):
         )
 
     def __len__(self):
-        return len(self.parts)
+        return len(self.part_ids)
 
     def __iter__(self):
-        for i in range(len(self)):
+        for i in self.part_ids:
             yield (self.name, i)
 
+    def get_external_dependencies(self, all_hlg_keys):
+        return set()
+
+    def cull(self, keys):
+        return ParquetSubgraph(
+            name=self.name,
+            engine=self.engine,
+            fs=self.fs,
+            meta=self.meta,
+            columns=self.columns,
+            index=self.index,
+            parts=self.parts,
+            kwargs=self.kwargs,
+            part_ids=[i for i in self.part_ids if (self.name, i) in keys],
+        )
```

Comment:
It's probably not an issue in practice, but this is maybe O(n^2).  We might want to store `self.part_ids` as a set for now?

Reply:
That was also my first thought, but I am wondering if some of the optimizations depend on the order of the parts?
If not I agree, a set make more sense.

Reply:
I don't know of any currently.  It's probably cheaper to sort on an as-needed basis though than do what we are here though.